addons/account/models/account_payment_method.py

```diff
@@ -121,6 +121,19 @@ class AccountPaymentMethodLine(models.Model):
     payment_type = fields.Selection(related='payment_method_id.payment_type')
     company_id = fields.Many2one(related='journal_id.company_id')
     available_payment_method_ids = fields.Many2many(related='journal_id.available_payment_method_ids')
+    payment_sequence = fields.Boolean(
+        string='Dedicated Payment Sequence',
+        compute='_compute_payment_sequence', readonly=False, store=True, precompute=True,
+        help="Check this box if you don't want to share the same sequence on payments and bank transactions posted on this journal",
+    )
+
+    @api.depends('journal_id.type')
+    def _compute_payment_sequence(self):
+        for method in self:
+            if method.journal_id.type in ('bank', 'cash', 'credit'):
```

Comment:
Isn't this always the case for payment methods?

Reply:
I have not found any constrains on the `journal_id` field, is there ? 
Hence, it is theoretically possible to have a journal type other than 'cash', 'credit and 'bank' set.

Reply:
There is no constraint, but it's not accessible from the UI, and never read from the code for sale/purchase/misc journals
```xml
<page id="inbound_payment_settings" string="Incoming Payments" name="page_incoming_payments" invisible="type not in ['cash', 'bank']">
```
(note that `credit` doesn't exist yet :wink:)